### PR TITLE
fix(legacy-charts): tune chartoptions type check

### PR DESCRIPTION
**Pull request by thabarbados**
### Fix or Enhancement?
tune legacy charts options type check

- [x] All tests passed

### Environment
- OS: BigSur MacOs
- NPM Version: 8.5.1

### CODE REVIEW

File: legacy/src/Charts.js [hunk 1]
```diff
@@ -74,6 +74,7 @@ export function generateChart(chartId, chartType, chartController) {
     computed: {
       hasAnnotationPlugin() {
```
Comment (dangreen) [inline legacy/src/Charts.js:75]:
```suggestion
      hasAnnotationPlugin() {
        return this.chartOptions?.plugins?.hasOwnProperty(ANNOTATION_PLUGIN_KEY)
```

maybe better like that?

Comment (thabarbados) [reply legacy/src/Charts.js:75]: @dangreen `Do not access Object.prototype method 'hasOwnProperty' from target object.`

Comment (dangreen) [reply legacy/src/Charts.js:75]: @thabarbados 
```
Reflect.hasOwnProperty(this.chartOptions?.plugins, ANNOTATION_PLUGIN_KEY)
```

Comment (thabarbados) [reply legacy/src/Charts.js:75]: @dangreen has the same error((

Comment (dangreen) [reply legacy/src/Charts.js:75]: @thabarbados 

```js
const plugins = this.chartOptions?.plugins

return plugins && Reflect.has(plugins, ANNOTATION_PLUGIN_KEY)
```

Comment (thabarbados) [reply legacy/src/Charts.js:75]: @dangreen  has Vue error in build version`[Vue warn]: Property or method "chartOptions" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property. See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.`

Comment (dangreen) [reply legacy/src/Charts.js:75]: @thabarbados ok. please find short variant of current code, cause current variant is too long

Comment (thabarbados) [reply legacy/src/Charts.js:75]: @dangreen fixed

File: legacy/src/Charts.js [hunk 2]
```diff
@@ -73,12 +73,10 @@ export function generateChart(chartId, chartType, chartController) {
...
-        return (
-          Object.keys(this.chartOptions).length > 0 &&
-          'plugins' in this.chartOptions &&
-          Object.keys(this.chartOptions.plugins).length > 0 &&
-          ANNOTATION_PLUGIN_KEY in this.chartOptions.plugins
-        )
+        const pluginSettings =
+          this.chartOptions?.plugins?.[ANNOTATION_PLUGIN_KEY]
+
+        return pluginSettings !== undefined
```
Comment (dangreen) [inline legacy/src/Charts.js:79]:
```suggestion
        return typeof pluginSettings !== 'undefined'
```

Comment (dangreen) [inline legacy/src/Charts.js:79]: or just 
```suggestion
        return Boolen(this.chartOptions?.plugins?.[ANNOTATION_PLUGIN_KEY])
```

Comment (thabarbados) [reply legacy/src/Charts.js]: @dangreen Used this option. I don't like Boolean)

**Review by dangreen — APPROVED** [review]

**Pull request merged**